hack/debug-network.sh

```diff
@@ -0,0 +1,574 @@
+#!/bin/bash
+
+# echoes args to stderr and exits
+die () {
+    echo "$*" 1>&2
+    exit 1
+}
```

Comment:
Since these could be `source`d more than once due to the `hack/lib/init.sh` logic, we like
```sh
readonly -f funcname
```

to ensure there's no chance of having different versions of a function active during the lifetime of a script